nexus/src/app/background/driver.rs

```diff
@@ -227,26 +231,39 @@ impl Drop for Driver {
 /// corresponding task has been created and then wired up with just an
 /// `&Activator` (not a `&mut Activator`).  See the [`super::init`] module-level
 /// documentation for more on why.
-pub struct Activator {
+#[derive(Clone)]
+pub struct Activator(Arc<ActivatorInner>);
```

Comment:
Personally I kind of prefer to make this more explicit among users (so that it would _not_ impl `Clone` but instead they'd use `Arc<Activator>`.  The reason is that I'm not generally sure what the semantics of `Clone` are going to be for something like this -- is it a thin handle to the same underlying thing or am I getting a "new" thing (and has the implementor actually considered it?), whereas with `Arc<Activator>` this behavior is clear and the implementor's choice doesn't matter.

I think my preference is probably less idiomatic than what you've got here though.

Reply:
Yeah, I dunno. I'm sensitive to the concern about making it clear what kind of cloning you get when you clone something...but I also feel like internally-`Arc`ed types that conceal the `Arc` from the API consumer are so widespread in the Rust ecosystem that trying to be explicit about it is kind of fighting a losing battle, and maybe it's better to just follow the broader ecosystem (and it's nice to not force everyone who uses the thing to not have to import `Arc`, I guess?).

If you feel strongly about it, I'm happy to change this!